scipy/sparse/_csr.py

```diff
@@ -122,6 +122,47 @@ def tobsr(self, blocksize=None, copy=True):
 
     tobsr.__doc__ = _spbase.tobsr.__doc__
 
+
+    def _broadcast_to(self, shape, copy=False):
+
+        old_shape = self.shape
+        
+        if old_shape == shape:
+            return self.copy() if copy else self
+        
+        if len(shape) != 2:
+            raise ValueError("Target shape must be a tuple of length 2.")
+
+        # Ensure the old shape can be broadcast to the new shape
+        if any((o != 1 and o != n) for o, n in zip(old_shape, shape)):
+            raise ValueError(f'current shape {old_shape} cannot be'
+                             f' broadcast to new shape {shape}')
+        
+        if self.nnz == 0: # array has no non zero elements
+            return self.__class__(np.zeros(shape, dtype=self.dtype), copy=False)
```

Comment:
We can use the constructor that takes a shape directly, no need to create a dense array of zeros here.

Reply:
Done at a41da4b0720c0c1f44278fffa1ed133cbae9cb76.